rtil/src/threads/ue/rebo/livesplit.rs

```diff
@@ -97,4 +125,48 @@ impl Run {
     pub fn attempt_count() -> u32 {
         LIVESPLIT_STATE.lock().unwrap().timer.run().clone().attempt_count()
     }
+    pub fn save_splits(path: impl AsRef<Path>) -> Result<(), SplitsSaveError> {
+        let state = LIVESPLIT_STATE.lock().unwrap();
+        let path = path.as_ref();
+        let filename = path
+            .file_name().expect("Could not get filename")
+            .to_str().expect("Could not convert filename to &str")
+            .to_owned();
+        let filename = sanitize_filename::sanitize(filename);
+        let valid_splits = VALID_SPLITS.lock().unwrap();
+        let joined_filename = LiveSplit::splits_path().join(&filename);
+        let path = match (path.is_relative(), valid_splits.contains(path)) {
+            (true, true) => joined_filename,
+            (true, false) => joined_filename,
```

Comment:
Can be combined using a wildcard: `(true, _) => ...` which also allows inlining `joined_filename` such that it only needs to be done if its case occurs.